### ISSUE TITLE
IDEX Printer, setting Hotend Count and Extruder Count to 2 is not working #2841

### ISSUE THREAD

**Issue by badonj001** [bigtreetech/BIGTREETECH-TouchScreenFirmware#2842]
This question is the BUG!
https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/issues/2841

**Comment by badonj001** [comment]
So to sum I will list the bugs surrounding the config.ini settings.

As indicated in the congif.ini all of the following settings are active in Marlin:
# General options which MUST be always activated:
#   EEPROM_SETTINGS (in Configuration.h)
#   BABYSTEPPING (in Configuration_adv.h)
#   AUTO_REPORT_TEMPERATURES (in Configuration_adv.h)
#   AUTO_REPORT_POSITION (in Configuration_adv.h)
#   EXTENDED_CAPABILITIES_REPORT (in Configuration_adv.h)
#   M115_GEOMETRY_REPORT (in Configuration_adv.h)
#   M114_DETAIL (in Configuration_adv.h)
#   REPORT_FAN_CHANGE (in Configuration_adv.h)


Serial Port Speed when set to anything other than 115200: 6, still sets it to 115200 as in marlin when I change #define BAUDRATE_2 250000, the serial port loses connectivity. Updating Marlin back to 115200 makes TFT  connect and work again.
SO: THIS SETTING DOES NOT WORK: "serial_port:P1:7 P2:0 P3:0 P4:0"

Extruder, Hotend Settings do not load when the configuration is reset.txt, all commits were removed and still it ignores these settings (These setting do not work):
hotend_count:2
heated_bed:1
heated_chamber:0
ext_count:2
fan_count:2
controller_fan:1
max_temp:T0:500 T1:500 T2:500 T3:500 T4:500 T5:500 BED:150 CHAMBER:60
min_temp:180
fan_max:F0:255 F1:255 F2:255 F3:255 F4:255 F5:255 CtA:255 CtI:255
size_min:X0 Y0 Z0
size_max:X310 Y310 Z350

SO: All of These setting also do not work from the config.ini.

#   Options: [OFF: 0, POPUP: 1, TOAST: 2]
#     OFF:   No notification. The message is ignored.
#     POPUP: Display a popup window for user confirmation.
#     TOAST: A non-blocking Toast notification is displayed for few seconds. No user interaction is needed.
ack_notification:2

#### Knob LED Color (only for TFT28/TFT35_E3/TFT43/TFT50/TFT70 V3.0)
# Knob LED color at startup.
#   Options: [OFF: 0, WHITE: 1, RED: 2, ORANGE: 3, YELLOW: 4, GREEN: 5, BLUE: 6, INDIGO: 7, VIOLET: 8]
knob_led_color:6

If I use anything other than what is already here, I get errors when config is updating:

#--------------------------------------------------------------------
# Start, End & Cancel G-code Commands
#--------------------------------------------------------------------

# NOTES for users having a filament sensor connected to the mainboard:
#   1) Enable the start/end G-code below.
#   2) Add the following commands to the start/end G-code:
#      - start_gcode: M75\n
#      - end_gcode: M77\n

#### Start G-code Status
#   Options: [disable: 0, enable: 1]
start_gcode_enabled:0

#### End G-code Status
#   Options: [disable: 0, enable: 1]
end_gcode_enabled:0

#### Cancel G-code Status
#   Options: [disable: 0, enable: 1]
cancel_gcode_enabled:0

#### Start G-code
# This G-code will run before starting a print if "start_gcode_enabled" is enabled.
#   Value range: [min: 3, max: 75 characters]
start_gcode:G28 XY R20\n

#### End G-code
# This G-code will run after a print is completed if "end_gcode_enabled" is enabled.
#   Value range: [min: 3, max: 75 characters]
end_gcode:M104 S0\nM140 S0\nM107\nM18\n

#### Cancel G-code
# This G-code will run when a print is cancelled if "cancel_gcode_enabled" is enabled.
#   Value range: [min: 3, max: 75 characters]
cancel_gcode:M104 S0\nM140 S0\nG28 XY R10\nM107\nM18\n


Go here https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/issues/2841 for all of the required information for this bug reporting. Too many things do not get set per the config.cfg that I know it is not me. I removed all commits and still does not work right?

**Comment by badonj001** [comment]
The problem was as is sometimes the case operator headspace,lol. My problem is relying on my memory from about 7-months ago when I first configured one of these display for 3d printer donation. I do not remember why, but I had to put an empty text file on the sdcard to reset the firmware. So apparently after reading the instructions again I noticed that the reset.txt is the absolute last thing the firmwares looks for after updating everything else. So It was loading the config settings I set, then running the reset undoing all of the settings it just changed. 

https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware#resetting-to-default-settings
